docs/performance/inserts/sequences.rst

```diff
@@ -0,0 +1,182 @@
+.. _autogenerated_sequences_performance:
+
+###########################################################
+ Autogenerated sequences and PRIMARY KEY values in CrateDB
+###########################################################
+
+If you are coming to CrateDB from systems like PostgreSQL or MySQL one thing you
+may notice is that CrateDB does not have a built-in auto-incrementing "serial"
+data type. This is because CrateDB is a distributed database, designed to scale
+horizontally, and this is achieved by making so that as many operations as
+possible can be completed by each node independently, without having to
+coordinate with other nodes. Maintaining an auto-increment value would mean that
+nodes would need to check with each other before allocating a new value to a
+row, creating a bottleneck and hindering our ability to achieve `extremely fast
+ingestion speeds`_.
+
+That said, there are many alternatives available and we can also implement true
+consistent/synchronized sequences if we want to.
+
+************************************
+ Using a timestamp as a primary key
+************************************
+
+This option involves declaring a column as follows:
+
+.. code:: psql
+
+   BIGINT DEFAULT now() PRIMARY KEY
+
+**Pros:** Always increasing number - ideal if we need to timestamp records
+creation anyway
+
+**Cons:** gaps between the numbers, not suitable if we may have more than one
+record on the same millisecond
+
+*************
+ Using UUIDs
+*************
+
+This option involves declaring a column as follows:
+
+.. code:: psql
+
+   TEXT DEFAULT gen_random_text_uuid() PRIMARY KEY
+
+**Pros:** Globally unique, no risk of conflicts if merging things from different
+tables/environments
+
+**Cons:** No order guarantee. Not as human-friendly as numbers. String format may
+not be applicable to cover all scenarios.
```

Comment:
Additionally, I think it's worth adding that no range queries are possible, as UUIDs aren't sequential. This is one of the main benefits of algorithms like a [snowflake_id](https://en.wikipedia.org/wiki/Snowflake_ID)